pym/bob/generators/QtCreatorGenerator.py

```diff
@@ -79,12 +81,16 @@ def addRunSteps(outFile, runTargets):
 
 def addBuildConfig(outFile, num, name, buildArgs, buildMeFile):
     outFile.write('<valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.' + str(num) + '">\n')
-    outFile.write('<value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">' + os.getcwd() + '</value>\n')
+    outFile.write('<value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">' + (os.getcwd() if not isWindows() else os.popen('pwd -W').read().strip()) + '</value>\n')
     outFile.write('<valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">\n')
     outFile.write(' <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">\n')
     outFile.write('  <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>\n')
-    outFile.write('  <value type="QString" key="ProjectExplorer.ProcessStep.Arguments">' + buildArgs + '</value>\n')
-    outFile.write('  <value type="QString" key="ProjectExplorer.ProcessStep.Command">' + buildMeFile + '</value>\n')
+    if isWindows():
+        outFile.write('  <value type="QString" key="ProjectExplorer.ProcessStep.Arguments">-msys2 -defterm -no-start -use-full-path -here -c "PATH=' + os.getenv('PATH') + ' sh ' + buildMeFile + ' -v ' + buildArgs + '"</value>\n')
+        outFile.write('  <value type="QString" key="ProjectExplorer.ProcessStep.Command">' + os.path.normpath(os.path.join(os.getenv('WD').replace('\\', '/'), '..', '..', 'msys2_shell.cmd')) + '</value>\n')
```

Comment:
That line feels magic. What's in `WD` and is the relative path to `msys2_shell.cmd` always the same? I'm just curious...

Reply:
that depends on this: https://github.com/msys2/MSYS2-packages/blob/master/filesystem/msys2_shell.cmd
It is important to start MSYS2 with this msys2_shell.cmd. I don't know a better solution.
I should log out an error, if WD is not set, with the information, that the user has to start MSYS2 by this script?!

Reply:
I think you must test it in advance. Because `os.getenv()` returns `None` if the variable is not set and then the code will crash. Better raise a `BuildError` with a proper error message before.